docs/extended_api/asynchronous_operations_library/aligned_size_t.md

```diff
@@ -0,0 +1,61 @@
+---
+grand_parent: Extended API
+parent: Asynchronous operations library
+---
+
+# cuda::**aligned_size_t**
+
+Defined in header [`<cuda/barrier>`](../headers/barrier.md)
+
+Defined in header [`<cuda/pipeline>`](../headers/pipeline.md)
+
+```c++
+template<size_t Alignment>
+struct aligned_size_t;
+```
+
+The class template `cuda::aligned_size_t` is a _shape_ representing an extent of bytes with a statically defined (address and size) alignment.
+
+## Template parameters
+
+| Alignment | the address & size alignement of the byte extent |
+
+## Data members
+
+| [align](./aligned_size_t/align.md) | the alignment of the byte extent |
+| [value](./aligned_size_t/value.md) | the size of the byte extent      |
+
+## Member functions
+
+| [(constructor)](./aligned_size_t/constructor.md) | constructs an _aligned size_                                                      |
+| (destructor) [implicitly declared]               | trivial implicit destructor                                                       |
+| operator= [implicitly declared]                  | trivial implicit copy/move assignment                                             |
+| operator size_t                                  | implicit conversion to [`size_t`](https://en.cppreference.com/w/cpp/types/size_t) |
+
+## Notes
+
+If `value` is not a multiple of `align` the behavior is undefined.

```

Comment:
Is `align=16` and `value=0` ok ? (e.g. does this support sizes that are smaller than the alignment)

Maybe:

```suggestion
If `value` is smaller than `align` or `value` is not a multiple of `align`, the behavior is undefined.
```

Reply:
No, memcpy_async is explicitly UB when the size is 0. The wording here is fine, since it does allow zero as is (and the suggested change would allow, say, 7, for alignment of 8, which is not intended).

Reply:
> No, memcpy_async is explicitly UB when the size is 0. 

Where is this stated? 

In the docs of `memcpy_async` below it says:

>  If either `destination` or `source` is an invalid or null pointer, the behavior is undefined (even if `count` is zero).

That suggests that a `count == 0` is ok as long as `destination` and `source` are valid.

Reply:
Whoops, I got carried away. That'll teach me to not respond when it's way too late for me to be awake anyway :P 

Anyway, the second part stands.

Reply:
@c0riolis this issue can be marked as resolved.